x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/allocation/SearchableSnapshotDiskThresholdIntegTests.java

```diff
@@ -268,6 +271,97 @@ public void testHighWatermarkCanBeExceededOnColdOrFrozenNode() throws Exception
         });
     }
 
+    public void testOvercommitInitializingSearchableSnapshotShards() throws Exception {
```

Comment:
Can we use a similar name to the previous test, since they sort of test the same but in different flavors?

Reply:
Also, I wonder why the tests are structured differently. The previous one mounts the same indices twice under different names. This one picks one not to mount, mounts the rest and then mounts the one extra index. I do not think there is a reason for that difference and maybe we can share most of the code between the two tests if we align the tests?

Reply:
I've extracted some common code of the tests into methods:

* createIndices
* createRepository
* createSnapshot
* mountIndices